src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java

```diff
@@ -1,54 +1,157 @@
 package org.folio.circulation.domain.notice;
 
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 import org.folio.circulation.domain.Item;
 import org.folio.circulation.domain.Loan;
 import org.folio.circulation.domain.Request;
+import org.folio.circulation.domain.RequestType;
+import org.folio.circulation.domain.ServicePoint;
 import org.folio.circulation.domain.User;
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.support.JsonArrayHelper;
+import org.folio.circulation.support.JsonStringArrayHelper;
+import org.joda.time.DateTime;
 import org.joda.time.DateTimeZone;
 
 import io.vertx.core.json.JsonObject;
 
 public class NoticeContextUtil {
 
-  private static final String PATRON = "patron";
+  private static final String USER = "user";
   private static final String ITEM = "item";
+  private static final String REQUEST = "request";
+  private static final String LOAN = "loan";
+
+  private static final String UNLIMITED = "unlimited";
 
   private NoticeContextUtil() {
   }
 
-  public static JsonObject createLoanNoticeContext(Loan loan) {
-    return createLoanNoticeContext(loan, DateTimeZone.UTC);
+  public static JsonObject createLoanNoticeContext(Loan loan, LoanPolicy loanPolicy) {
+    return createLoanNoticeContext(loan,loanPolicy, DateTimeZone.UTC);
   }
 
-  public static JsonObject createLoanNoticeContext(Loan loan, DateTimeZone timeZone) {
-    return createNoticeContextFromItemAndPatron(loan.getItem(), loan.getUser())
-      .put("dueDate", loan.getDueDate().withZone(timeZone).toString());
+  public static JsonObject createLoanNoticeContext(
+    Loan loan, LoanPolicy loanPolicy, DateTimeZone timeZone) {
+
+    return new JsonObject()
+      .put(USER, createPatronContext(loan.getUser()))
```

Comment:
Given the grouping has been renamed from patron to user, would it be more consistent to rename the `createPatronContext` method to `createUserContext`?

Reply:
Done